export/mlab_export.py

```diff
@@ -0,0 +1,591 @@
+#!/usr/bin/python
+# Copyright 2014 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Summary:
+  mlab_export.py serializes RRD data collected by the M-Lab, collectd plugin.
+
+  mlab_export.py supports many options for use by a human operator. However,
+  typical usage should be automated using something like crond. Because typical
+  usage is automated, all parameters have sensible defaults, tuned to running on
+  M-Lab.
+
+Overview of default operation:
+  * Script finds all RRD files under --rrddir_prefix.
+  * Script opens a file for output under --output_dir (optionally compressed).
+  * Script reads values from each RRD between --ts_start and --ts_end.
+  * Script determines ts_start and ts_end from 'mtime' on LAST_EXPORT_FILENAME.
+    - On first run, LAST_EXPORT_FILENAME is created. ts_end is set to nearest
+      hour from current time, and ts_start is set to ts_end minus --length.
+    - On later runs, ts_start is set to the 'mtime' of LAST_EXPORT_FILENAME.
+      And, ts_end is set to the most recent hour.
+  * Script only exports metrics defined in the export_metrics.conf config file.
+  * On success, script sets mtime on LAST_EXPORT_FILENAME to ts_end, in
+    preparation for the next run.
+
+Examples:
+  # Default operation should not require additional parameters.
+  ./mlab-export.py
```

Comment:
Ah, yes. Done. I've renamed mlab_export.py in the examples. 55c26ccbd1ae800dc3d2ef8fa6c662840262269e